### ISSUE TITLE
ES6 rest args triggers warnings

### ISSUE THREAD

**Issue by zpao** [eslint/eslint#1543]
This code
```js
"use strict";
module.exports.b = function foo(...args) {
  return args;
};
```

With this config
```yaml
settings:
  ecmascript: 6
env:
  node: true
rules:
  block-scoped-var: 1
```

Results in this
```
eslint-test.js
  2:35  warning  "args" used outside of binding context  block-scoped-var
  3:9   warning  "args" used outside of binding context  block-scoped-var
  3:9   error    'args' is not defined                   no-undef

✖ 3 problems
```

The obvious problem here is that rest args just aren't supported in these rules. (v0.11.0-alpha.0)

**Comment by nzakas** [comment]
Just took another look. Seems like escope is still confused about this, I'll followup.

**Comment by Willibaur** [comment]
Hi all,
I've tried to make this bit of code good enough for ESlint rule `prefer-rest-params` but I ran out of ideas. 
The problem is with the line `this._super(...arguments);`
Any help would be appreciated. Thanks
```
didTransition() {
    this._super(...arguments);
    Ember.run.scheduleOnce('afterRender', this, () => {
      this._trackPage(this.get('url'), document.title);
    });
  },
```

**Comment by platinumazure** [comment]
@Willibaur The spread operator needs to be used on a defined variable. Usually, you would do so by declaring a rest parameter in the method definition.

Try this:

```js
  didTransition(...args) { // <-- Variable definition: args
    this._super(...args);
    Ember.run.scheduleOnce('afterRender', this, () => {
      this._trackPage(this.get('url'), document.title);
    });
  },
```

**Comment by Willibaur** [comment]
Awesome, thanks @platinumazure it worked straight away, thank you very much for you quick answer
